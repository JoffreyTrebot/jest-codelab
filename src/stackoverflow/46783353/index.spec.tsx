import React from 'react';
import TestRenderer from 'react-test-renderer';
import { App } from './';

describe('App', () => {
  it('t1', () => {
    const testRenderer = TestRenderer.create(<App></App>);
    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();

    if (tree) {
      tree.props.onKeyDown({ keyCode: 37 });
    }
    tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();

    if (tree) {
      tree.props.onKeyDown({ keyCode: 38 });
    }
    tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();

    if (tree) {
      tree.props.onKeyDown({ keyCode: 39 });
    }
    tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();

    if (tree) {
      tree.props.onKeyDown({ keyCode: 40 });
    }
    tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
