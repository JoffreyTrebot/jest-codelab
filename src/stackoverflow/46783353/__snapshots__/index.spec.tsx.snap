// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App t1 1`] = `
<div
  onKeyDown={[Function]}
>
  x:
  0
  ,y:
  0
</div>
`;

exports[`App t1 2`] = `
<div
  onKeyDown={[Function]}
>
  x:
  -1
  ,y:
  0
</div>
`;

exports[`App t1 3`] = `
<div
  onKeyDown={[Function]}
>
  x:
  -1
  ,y:
  1
</div>
`;

exports[`App t1 4`] = `
<div
  onKeyDown={[Function]}
>
  x:
  0
  ,y:
  1
</div>
`;

exports[`App t1 5`] = `
<div
  onKeyDown={[Function]}
>
  x:
  0
  ,y:
  0
</div>
`;
